e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/filter-payment-details.spec.js

```diff
@@ -26,52 +28,693 @@ context(`users can filter payment details by facility id and payment reference a
+        .withId(12)
+        .withAmount(200)
+        .withReference('Second payment ref')
+        .withFeeRecords([secondFeeRecord])
+        .build();
+      const thirdPayment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+        .withId(13)
+        .withAmount(300)
+        .withReference('Third payment ref')
+        .withFeeRecords([thirdFeeRecord])
+        .build();
+      const fourthPayment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+        .withId(14)
+        .withAmount(400)
+        .withReference('Another ref')
+        .withFeeRecords([thirdFeeRecord, fourthFeeRecord])
+        .build();
+
+      cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord, thirdFeeRecord, fourthFeeRecord]);
+      cy.task(NODE_TASKS.INSERT_PAYMENTS_INTO_DB, [firstPayment, secondPayment, thirdPayment, fourthPayment]);
+
+      pages.landingPage.visit();
+      cy.login(USERS.PDC_RECONCILE);
+
+      cy.visit(`/utilisation-reports/${reportId}`);
+
+      pages.utilisationReportPage.paymentDetailsTabLink().click();
+
+      pages.utilisationReportPage.paymentDetailsTab.filters.paymentCurrencyRadioInput(paymentCurrencyFilter).click();
+      cy.keyboardInput(pages.utilisationReportPage.paymentDetailsTab.filters.paymentReferenceInput(), partialPaymentReferenceFilter);
+      cy.keyboardInput(pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdInput(), partialFacilityIdFilter);
+
+      pages.utilisationReportPage.paymentDetailsTab.filters.submitButton().click();
+
+      cy.url().should(
+        'eq',
+        relative(
+          `/utilisation-reports/${reportId}?paymentDetailsPaymentCurrency=${paymentCurrencyFilter}&paymentDetailsPaymentReference=${partialPaymentReferenceFilter}&paymentDetailsFacilityId=${partialFacilityIdFilter}#payment-details`,
+        ),
+      );
+
+      pages.utilisationReportPage.paymentDetailsTab.filters.paymentCurrencyRadioInput(paymentCurrencyFilter).should('be.checked');
+      pages.utilisationReportPage.paymentDetailsTab.filters.paymentReferenceInput().should('have.value', partialPaymentReferenceFilter);
+      pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdInput().should('have.value', partialFacilityIdFilter);
+
+      pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(firstPayment.id, firstFeeRecord.id).should('exist');
+      pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(secondPayment.id, secondFeeRecord.id).should('not.exist');
+      pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(thirdPayment.id, thirdFeeRecord.id).should('exist');
+      pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(fourthPayment.id, thirdFeeRecord.id).should('not.exist');
+      pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(fourthPayment.id, fourthFeeRecord.id).should('not.exist');
+    });
```

Comment:
This test is huge, just over 700 lines 🤯 

Can we split it up please? 

How long does it take to run?

Reply:
It was around 33s, but I agree in retrospect it should've been split up as it's rather large. I've split this by filter, and kept a single file for generic checks.